virttest/qemu_vm.py

```diff
@@ -455,29 +455,30 @@ def add_qmp_monitor(devices, monitor_name, filename):
             return cmd
 
         def add_serial(devices, name, filename):
+            # Arm lists "isa-serial" as supported but can't use it,
+            # fallback to "-serial"
+            legacy_cmd = " -serial unix:'%s',server,nowait" % filename
             if (not devices.has_option("chardev") or
-                    not any(devices.has_device(dev)
-                            for dev in ("isa-serial", "sclpconsole", "spapr-vty"))):
-                return " -serial unix:'%s',server,nowait" % filename
+                    'arm' in params.get("machine_type", "")):
+                return legacy_cmd
+            for serial_dev in ("sclpconsole", "spapr-vty", "isa-serial"):
+                if devices.has_device(serial_dev):
+                    break
+            else:
```

Comment:
Is this `else` in correct indent?

Reply:
@ldoktor I am trying to refactor the whole serial creation part, since it currently does not support other chardev backend(tcp, pty alike), and neither support Q35 hotplug(for no definition of the bus virt-serial-pci). There I totally drop this `add_serial()` funcion, see https://github.com/avocado-framework/avocado-vt/pull/1639